### PR TITLE
Accept list parameter in new_sets union

### PR DESCRIPTION
**Pull request by mjduijn**
The new_sets union does not accept a list as one of its parameters, leading to a number of build breakages when updating to Skylib 0.9.0, see also https://github.com/bazelbuild/bazel-skylib/issues/163#issuecomment-511408099.
The function [documentation](https://github.com/bazelbuild/bazel-skylib/blob/2b38b2f8bd4b8603d610cfc651fcbb299498147f/lib/new_sets.bzl#L179) states that it should accept an arbitrary number of sets or lists, so the obvious fix was to make it do so :)

### CODE REVIEW

File: lib/new_sets.bzl [hunk 1]
```diff
@@ -22,6 +22,8 @@
 
 load(":dicts.bzl", "dicts")
 
+_list_type = type([])
```
Comment (techsy730) [inline lib/new_sets.bzl]: load "types.bzl" and use "types.is_list()"?

**Comment by aiuto** [comment]
@c-parsons Can you take a look at this. I was about to make the same change. It will help rules_go and rules_docker migrate to the current skylib. Otherwise they are going to have to use old_sets.bzl until they can make the bigger change.

**Comment by mjduijn** [comment]
Thanks for the feedback, I'll be back from vacation this weekend and have another look at this then

**Comment by mjduijn** [comment]
I'm not sure what to do about the build failure, I'm guessing it has to do with #184.
Some help would be appreciated.

**Comment by aiuto** [comment]
You need to add types as a dep of new_sets.  Try this patch

index 8ae8259c5a..779812ca32 100644
--- a/lib/BUILD
+++ b/lib/BUILD
@@ -40,7 +40,10 @@ bzl_library(
 bzl_library(
     name = "new_sets",
     srcs = ["new_sets.bzl"],
-    deps = [":dicts"],
+    deps = [
+        ":dicts",
+        ":types",
+    ],
 )
 
 bzl_library(

**Comment by aiuto** [comment]
Also you can't just build the docs (which seem to be failing) because of strangeness with how repositories get named.  Try this:
        bazel build docs:* --incompatible_remap_main_repo

**Comment by mjduijn** [comment]
Thanks, that fixed it

**Comment by techsy730** [comment]
Maybe a `to_sets` or similar parameter would be more general here?
Would turn list elements (or maybe tuples too, or even any iterable) into sets, and leave sets as is.

Yes, that basically is what `[x if types.is_set(x) else sets.make(x) for x in things_lists_list]` does, but perhaps maybe a brief survey can be done whether this is a common enough comprehension to pull into a method?

**Comment by mjduijn** [comment]
Thank you for the feedback. As I have not worked intensively with Bazel and Skylib for over a year I have no idea if the referenced issue is still relevant and how best to fix it.

If it helps the discussion I can keep this PR open but I will not be make any further changes to it.

**Comment by tetromino** [comment]
Ok. Closing for now, since afaik the build breakage has been already fixed elsewhere. But we would welcome a PR that addresses https://github.com/bazelbuild/bazel-skylib/pull/179#pullrequestreview-625767179 :)

**Pull request closed without merging**